demo/app.py

```diff
@@ -23,6 +28,33 @@ def get_access():
 
 @app.route('/', methods=['GET', 'POST'])
 def index():
+    cas_url = app.config['CAS_URL']
+    service_url = app.config['SERVICE_URL']
+    authorized_users = app.config['AUTHORIZED_USERS']
+    logout_url = service_url+"logout"
+
+    if u'ticket' in request.args:
+        # got cas; validate ticket
+        try:
+            user = validate_cas(cas_url, request.args[u'ticket'], service_url)
+        except CASError as e:
+            return u'login failed (%s)' % str(e), 403
+        except Exception as e:
+            return u'login failed', 403
```

Comment:
Just my two cents. I will suggest using `'login failed {}'.format(403}` (`%` is also fine) to format your string. Using `,` to create spaces in a string make the code looks complicated than it actually is.

Reply:
`403` is the status code. It's not being displayed in the response body.